leshan-core/src/main/java/org/eclipse/leshan/core/request/argument/Arguments.java

```diff
@@ -0,0 +1,314 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Orange.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Orange - Add better support for Arguments of Execute Operation.
+ *******************************************************************************/
+package org.eclipse.leshan.core.request.argument;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.leshan.core.util.StringUtils;
+
+/**
+ * Arguments for Execute Operation.
+ */
+public class Arguments {
+
+    private final Map<Integer, Argument> argumentMap;
+
+    /**
+     * Arguments should be build using {@link ArgumentsBuilder}.
+     */
+    private Arguments(Map<Integer, Argument> argumentMap) {
+        this.argumentMap = argumentMap;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Arguments [argumentMap=%s]", argumentMap.toString());
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((argumentMap == null) ? 0 : argumentMap.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Arguments other = (Arguments) obj;
+        if (argumentMap == null) {
+            if (other.argumentMap != null)
+                return false;
+        } else if (!argumentMap.equals(other.argumentMap))
+            return false;
+        return true;
+    }
+
+    /**
+     * Returns number of arguments.
+     */
+    public int size() {
+        return argumentMap.size();
+    }
+
+    /**
+     * Check if Arguments has any elements.
+     */
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    /**
+     * Checks if  {@link Arguments} has {@link Argument} with digit.
+     */
+    public boolean hasDigit(int digit) {
+        for (Integer mapKey: argumentMap.keySet()) {
+            if (mapKey == digit) {
+                return true;
+            }
+        }
```

Comment:
Is there any reason to not use `argumentMap.containsKey` ?

Reply:
No reason, it just remainder of converting list to map. I wasn't focused enough.